src/device_queue.cc

```diff
@@ -48,26 +48,42 @@ namespace detail {
 			} else {
 				const auto host_cfg = cfg.get_host_config();
 
-				// Try to find a platform that can provide a unique device for each node.
-				bool found = false;
-				const auto platforms = cl::sycl::platform::get_platforms();
-				for(size_t i = 0; i < platforms.size(); ++i) {
-					auto&& platform = platforms[i];
-					const auto devices = platform.get_devices(cl::sycl::info::device_type::gpu);
-					if(devices.size() >= host_cfg.node_count) {
-						how_selected = fmt::format("automatically selected platform {}, device {}", i, host_cfg.local_rank);
-						device = devices[host_cfg.local_rank];
-						found = true;
-						break;
+				const auto try_find_device_per_node = [&host_cfg, &device, &how_selected](cl::sycl::info::device_type type) {
```

Comment:
I think it would be a bit nicer to factor this and the subsequent lambda out as separate functions (since the surrounding function is already quite huge), but it's not really necessary and maybe more in the scope of a refactoring PR.

Reply:
Yes, let's do it in a separate change (I've added it to the refactoring wishlist).